x-pack/plugins/session_view/public/components/ProcessTree/hooks.ts

```diff
@@ -110,20 +114,20 @@ class ProcessImpl implements Process {
 
 export const useProcessTree = ({
   sessionEntityId,
+  sessionLeaderEvent,
   forward,
   backward,
   searchQuery,
 }: UseProcessTreeDeps) => {
-  // initialize map, as well as a placeholder for session leader process
-  // we add a fake session leader event, sourced from wide event data.
-  // this is because we might not always have a session leader event
-  // especially if we are paging in reverse from deep within a large session
-  const fakeLeaderEvent = forward.find((event) => event.event.kind === EventKind.event);
+  // initialize map
   const sessionLeaderProcess = new ProcessImpl(sessionEntityId);
 
-  if (fakeLeaderEvent) {
-    fakeLeaderEvent.process = { ...fakeLeaderEvent.process, ...fakeLeaderEvent.process.entry };
-    sessionLeaderProcess.events.push(fakeLeaderEvent);
+  if (sessionLeaderEvent) {
```

Comment:
We aren't  guaranteed to even have a session leader event. Despite loading it before hand, there is still a chance it could be missing, so we should not make this a dependency on loading the process tree before, as we may not always have this information. Which is why I was sourcing it from the wide event of another event.

Reasons why it's missing could include: Filtering, sending up events for a session that started before the agent started.